### PR TITLE
Retention and callback url

### PR DESCRIPTION
**Pull request by tshaiman**
- extend the java sample to include retention Period in days and Callback URL.

### CODE REVIEW

File: API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java [hunk 1]
```diff
@@ -33,6 +33,13 @@ public class VideoIndexerClient {
     private static final String AccountName = "<Your_Account_Name_Here>";
     private static final String ApiVersion = "2022-08-01";
     private static final String ApiUrl = "https://api.videoindexer.ai";
+    
+    //If you want to be notified with POST events on your web site
```
Comment (ika-microsoft) [inline API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java:37]: to your web site

File: API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java [hunk 2]
```diff
@@ -33,6 +33,13 @@ public class VideoIndexerClient {
     private static final String AccountName = "<Your_Account_Name_Here>";
     private static final String ApiVersion = "2022-08-01";
     private static final String ApiUrl = "https://api.videoindexer.ai";
+    
+    //If you want to be notified with POST events on your web site
+    //The callback URL can contain additional query parameters for example adding the externalId field
+    //Or any Custom Field.
+    //Example Callback with custom Parameters : https://webhook.site/#!/0000/?externalId=1234&customField=MyCustomField
+    private static final String CallbackUrl ="" ;
```
Comment (ika-microsoft) [inline API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java:41]: remove extra space between quotes and semi column.

File: API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java [hunk 3]
```diff
@@ -101,7 +108,15 @@ public String uploadVideo(String videoUrl, String videoName) {
         map.put("privacy", "private");
         map.put("partition", "partition");
         map.put("videoUrl", videoUrl);
+        // For API Based Scenarios it is advised to set "NoStream" for faster indexing. 
         map.put("streamingPreset","NoStreaming");
+        //Retention Period of Video in days. Default is No retention. Max Allowed value is 7.
+        map.put("retentionPeriod","1");
+        // Use Callback URL to get notified on Video Indexing Events ( Start/ End Processing)
+        if (!CallbackUrl.isBlank()) {
+            map.put("callbackUrl", CallbackUrl);
```
Comment (ika-microsoft) [inline API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java:117]: Tomer, I think that the value should be encoded 
String encodedUrl = URLEncoder.encode(originalUrl, StandardCharsets.UTF_8);

This is also true for the videoUrl param to be honest.

**Review by ika-microsoft — APPROVED** [review]
Please see my comments before completing.

File: API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java (hunk 3 above)
Comment (tshaiman) [reply API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java:117]: Totally Agree ! 
thanks .

**Comment by tshaiman** [comment]
> Please see my comments before completing.

Done. great tips , thanks

**Comment by tshaiman** [comment]
close by Accident ;-(

**Pull request merged**